test/interface.spec.js

```diff
@@ -0,0 +1,179 @@
+    ]
+  })
+
+  tests.dht(defaultCommonFactory, {
+    skip: [
+      // dht.findpeer
+      // FIXME checking what is exactly go-ipfs returning
+      // https://github.com/ipfs/go-ipfs/issues/3862#issuecomment-294168090
+      'should fail to find other peer if peer does not exist',
+      // dht.findprovs
+      // FIXME go-ipfs endpoint doesn't conform with the others
+      // https://github.com/ipfs/go-ipfs/issues/5047
+      'should provide from one node and find it through another node',
+      // dht.get
+      // FIXME go-ipfs errors with  Error: key was not found (type 6)
+      // https://github.com/ipfs/go-ipfs/issues/3862
+      'should get a value after it was put on another node'
+    ]
+  })
+
+  tests.files(defaultCommonFactory, {
+    skip: [
+      // files.add
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should add a nested directory as array of tupples',
+      isNode ? null : 'should add a nested directory as array of tupples with progress',
+      // files.addPullStream
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should add pull stream of valid files and dirs',
+      // files.addReadableStream
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should add readable stream of valid files and dirs',
+      // files.catPullStream
+      // TODO not implemented in go-ipfs yet
+      'should export a chunk of a file',
+      'should export a chunk of a file in a Pull Stream',
+      'should export a chunk of a file in a Readable Stream',
+      // files.get
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should get a directory'
+    ]
+  })
+
+  tests.key(defaultCommonFactory, {
+    skip: [
+      // key.export
+      // TODO not implemented in go-ipfs yet
+      'export',
+      // key.import
+      // TODO not implemented in go-ipfs yet
+      'import'
+    ]
+  })
+
+  tests.ls(defaultCommonFactory, {
+    skip: [
+      // lsPullStream
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should pull stream ls with a base58 encoded CID',
+      // lsReadableStream
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should readable stream ls with a base58 encoded CID',
+      // ls
+      // FIXME https://github.com/ipfs/js-ipfs-api/issues/339
+      isNode ? null : 'should ls with a base58 encoded CID'
+    ]
+  })
+
+  tests.miscellaneous(defaultCommonFactory, {
+    skip: [
+      // stop
+      // FIXME go-ipfs returns an error https://github.com/ipfs/go-ipfs/issues/4078
+      'should stop the node'
+    ]
+  })
+
+  tests.object(defaultCommonFactory)
+
+  tests.pin(defaultCommonFactory)
+
+  tests.ping(defaultCommonFactory)
+
+  tests.pubsub(CommonFactory.create({
+    spawnOptions: {
+      args: ['--enable-pubsub-experiment'],
+      initOptions: { bits: 1024 }
```

Comment:
Why not use `1024` for all tests? It would make things way faster.

Reply:
They still are, the `defaultCommonFactory` passed to other suites uses 1024 by default https://github.com/ipfs/js-ipfs-api/pull/785/files/519bbf5c4374317d5355d9bddfb2fc5862ddd16d#diff-b4ff00e5aeab8b441e96f3d0b1082a2eR12

Reply:
Understood! Thanks :)